extensions/MasterMath/midi.js

```diff
@@ -0,0 +1,271 @@
+                menu: "pressedReleased",
+              },
+            },
+          },
+          {
+            opcode: "whenNote",
+            blockType: Scratch.BlockType.EVENT,
+            text: Scratch.translate("when note [note] [pressedReleased]"),
+            isEdgeActivated: false,
+            shouldRestartExistingThreads: true,
+            arguments: {
+              note: {
+                type: Scratch.ArgumentType.NOTE,
+                defaultValue: 60,
+              },
+              pressedReleased: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "pressed",
+                menu: "pressedReleased",
+              },
+            },
+          },
+          {
+            opcode: "noteOn",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: Scratch.translate("is note [note] on?"),
+            arguments: {
+              note: {
+                type: Scratch.ArgumentType.NOTE,
+                defaultValue: 60,
+              },
+            },
+          },
+          {
+            opcode: "noteVelocity",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("velocity of note [note]"),
+            arguments: {
+              note: {
+                type: Scratch.ArgumentType.NOTE,
+                defaultValue: 60,
+              },
+            },
+          },
+          {
+            opcode: "activeNotes",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("all active notes"),
+            disableMonitor: true,
+          },
+          {
+            opcode: "lastNote",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("last note [pressedReleased]"),
+            disableMonitor: true,
+            arguments: {
+              pressedReleased: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "pressed",
+                menu: "pressedReleased",
+              },
+            },
+          },
+        ],
+        menus: {
+          infoMenu: {
+            acceptReporters: false,
+            items: [
+              {
+                text: Scratch.translate("name"),
+                value: "name",
+              },
+              {
+                text: Scratch.translate("id"),
+                value: "id",
+              },
+            ],
+          },
+          pressedReleased: {
+            acceptReporters: false,
+            items: [
+              {
+                text: Scratch.translate("pressed"),
+                value: "pressed",
+              },
+              {
+                text: Scratch.translate("released"),
+                value: "released",
+              },
+            ],
+          },
+        },
+      };
+    }
+
+    MIDIinputDevices() {
+      return midiInputDevices;
+    }
+
+    midiDeviceInfo(args) {
+      if (midiInputDevices[args.number] != null) {
```

Comment:
Use `Scratch.Cast` on your arguments (mainly your non menu ones)
Since your only other inputs that are not menus are number inputs, use `Scratch.Cast.toNumber`
Ex.
`args.number = Scratch.Cast.toNumber(args.number)`